### PR TITLE
[PROTO-1433] Refactor audius-d to use cobra commands, make config kubectl-like

### PR DESCRIPTION
**Pull request by phelpsdb**
## Description

Cobra will allow us to better separate the app logic from the interface logic.

This PR piggybacks the config refactor being done by @alecsavvy and adds the ability to "context switch" between different configurations similar to how kubectl works.

Use `audius-d help [command]` to get info

Examples:
`audius-d config` - dump the current configuration to stdout
`audius-d config create-context mydevcontext -f path/to/template.toml` - make a new context from an existing template
`audius-d config get-context` - show the current context
`audus-d config set-context someothercontext` - switch contexts

Previous audius-d commands (up, devnet, down) work same before.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,323 +1,45 @@
...
-var nodeType string
-var seed bool
-var autoUpgrade bool
 
 // with the intent of reducing configuration,
 // the latest audius-docker-compose sha (from stage branch) is set at build time via ci.
 // this bakes the (tested) image dependency in, so we know that the built binary will always work.
 var imageTag string
 
 func main() {
```
Comment (phelpsdb) [inline main.go:22]: Everything below was moved to orchestration/docker.go

File: conf/context.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		}
+	}
+
+	var ctx ContextConfig
+	err = readConfigFromFile(contextFilePath, &ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ctx, nil
```
Comment (phelpsdb) [inline conf/context.go:59]: Is this just as bad in golang land as it is in C++ land? Should the function instead require the caller to pass in a pointer to their own ContextConfig?

File: orchestration/docker.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	"regexp"
+	"strings"
+
+	"github.com/joho/godotenv"
+)
+
+func runNodeDocker(nodeType string, network string, imageTag string, autoUpgrade bool) {
+	fmt.Printf("standing up %s on network %s\n", nodeType, network)
+	volumeFlag := ""
+	//volumeFlag = fmt.Sprintf("-v %s:/root/audius-docker-compose/%s/override.env", confFilePath, nodeType)
```
Comment (phelpsdb) [inline orchestration/docker.go:17]: This does not work currently with the switch to toml configs. Purposefully waiting for the config changes from @alecsavvy before integrating a solution.

File: conf/context.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+const (
+	ConfigVersion = "0.1"
+	ContextKey    = audiusContextKey("audius_cobra_context")
+)
+
+func ReadOrCreateContextConfig() (*ContextConfig, error) {
+	confDir, err := getConfigBaseDir()
+	if err != nil {
+		return nil, err
```
Comment (phelpsdb) [inline conf/context.go:21]: TODO: improve error handling to be consistent and more descriptive throughout the program

File: conf/context.go (hunk 2 above)
Comment (alecsavvy) [reply conf/context.go:59]: you mean have a function signature like
```
func ReadContextConfig(ctx *ContextConfig) (*ContextConfig, error) {
    // modify passed in ctx here and return?

}
```

Comment (alecsavvy) [reply conf/context.go:59]: similar to how the json marshal package works?

Comment (phelpsdb) [reply conf/context.go:59]: Looks like we can do it either way https://stackoverflow.com/questions/46987513/handling-dangling-pointers-in-go

**Comment by phelpsdb** [comment]
We will intentionally land this incomplete PR to provide a base for [this PR](https://github.com/AudiusProject/audius-d/pull/23) sooner than later. All that is missing to get passing tests is creating override.env files

**Review by alecsavvy — APPROVED** [review]

**Pull request merged**